File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
var kthDistinct = function(arr, k) {
  let count = 0
  for (let s of arr) {
    count += (arr.indexOf(s) == arr.lastIndexOf(s))
    if (count === k) return s
  }
  return ''
};

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.go
func kthDistinct(arr []string, k int) string {
  counter := make(map[string]int, len(arr))
  for _, v := range arr {
    counter[v]++
  }
  count := 1
  for _, v := range arr {
    if counter[v] == 1 {
      if count == k {
        return v
      }
      count ++
    }
  }
  return ""
}